### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by oacdutra**
Hi @dr-orlovsky,

This PR bump the  rust-amplify dependency to version 4.x.

### CODE REVIEW

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
Cool, but I think we must avoid using all these `.0` and `.as_inners()` since they will be very polluting the code downstream. Instead we need to add `#[wrapper(Deref)]` to all types deriving `Wrapper` in this library (that's the change in amplify 4.0: wrapper doesn't adds Deref impl by default and requires explicit flag for that).

File: bitcoin_scripts/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ exclude = []
 crate-type = ["lib", "staticlib"]
 
 [dependencies]
-amplify = "3.14.2"
+amplify = "4.0.0"
 bitcoin = "0.29.2" # We pin this version due to an upstream bug. See setails: <https://github.com/BP-WG/descriptor-wallet/pull/40>
 secp256k1 = { version = "0.24.2", features = ["global-context"] } # Remove after bitcoin #922
```
Comment (cryptoquick) [inline bitcoin_scripts/Cargo.toml:22]: It looks like all of these PRs have been merged:
https://github.com/rust-bitcoin/rust-bitcoin/pull/922#issuecomment-1084585501
I think this comment can be addressed, @dr-orlovsky? Can you provide more background on what this note means?

File: bitcoin_blockchain/Cargo.toml [hunk 2]
```diff
@@ -17,9 +17,9 @@ exclude = []
 crate-type = ["lib", "staticlib"]
 
 [dependencies]
-amplify = "3.14.2"
+amplify = "4.0.0"
 serde_crate = { package = "serde", version = "1", features = ["derive"], optional = true }
-chrono = "0.4.23"
+chrono = "0.4.24"
```
Comment (cryptoquick) [inline bitcoin_blockchain/Cargo.toml]: I think we change this to `chrono = { version = "0.4.24", default-features = false }`  as recommended here: https://github.com/chronotope/chrono/issues/602#issuecomment-1513829262

**Review by cryptoquick — CHANGES_REQUESTED** [review]
I've been doing some research into how to resolve [CVE-2020-26235](https://nvd.nist.gov/vuln/detail/CVE-2020-26235), the infamous time/chrono CVE, and it looks like this crate is one of the core places where this is depended on in the RGB ecosystem according to `cargo tree -i time@0.1.45` in `bitmask-core 0.6.0-beta.0`.

File: bitcoin_scripts/Cargo.toml (hunk 1 above)
Comment (dr-orlovsky) [reply bitcoin_scripts/Cargo.toml:22]: rust-bitcoin bug is still not fixed, but the pinning is not there anyway... And I am not sure why I decided that pinning was required the first place... So I think we can just remove the comment and that's it

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
My main comment was still not addressed: https://github.com/BP-WG/bitcoin_foundation/pull/18#pullrequestreview-1386422501

This PR also needs rebase

**Comment by oacdutra** [comment]
> My main comment was still not addressed: [#18 (review)](https://github.com/BP-WG/bitcoin_foundation/pull/18#pullrequestreview-1386422501)
> 
> This PR also needs rebase

I tried to merge and rebase to the master branch, but I encountered some challenges to make that. 

I saw the comments in PR #19, and I resolve to explain the goal to the branch:

Initially, I bumped amplify dependency in `descriptor-wallet` and `bp_foundation` two weeks ago (see [here](https://github.com/diba-io/bitmask-core/blob/4873ecce3dc5c429953579baebb3ebae781f763b/Cargo.toml#L108-L112)) to use `descriptor-wallet` in bitmask-core with WASM target.  Then, I opened this PR, and after approval, I would open the PR for `descriptor-wallet`.

I can close this PR and open another with only amplify bump. 


What do you think?

**Comment by dr-orlovsky** [comment]
@crisdut yes sure!

**Pull request closed without merging**